src/System.Linq.Expressions/src/System/Runtime/CompilerServices/CallSite.cs

```diff
@@ -567,7 +567,7 @@ private T CreateCustomUpdateDelegate(MethodInfo invoke)
                 Expression.Assign(rule, Expression.ArrayAccess(applicable, index))
             );
 
-            body.Add(Expression.Assign(index, Expression.Constant(0)));
+            body.Add(Expression.Assign(index, Utils.Constant(0)));
```

Comment:
Code behind custom delegate types never leaks to users, so uniqueness of references for these nodes is not needed.